### PR TITLE
pgsql: initial support

### PR DESCRIPTION
**Pull request by jufajardini**
- pgsql/parser: add startup message structure, initial nom parser and
corresponding unit test
- pgsql/mod: list pgsql modules that must be compiled
- lib: add pgsql to the module's list

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4241

Describe changes:
- pgsql/parser: add startup message structure, initial nom parser and
corresponding unit test
- pgsql/mod: list pgsql modules that must be compiled
- lib: add pgsql to the module's list

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

**Comment by victorjulien** [comment]
Please have a look at the failing QA tests, they show how the code can be tidied up a bit.

File: rust/src/pgsql/parser.rs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        })
+));
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+
+    #[test]
+    fn test_parse_pgsql_startup_packet() {
```
Comment (victorjulien) [inline rust/src/pgsql/parser.rs:65]: I think a good next step would be to add a invalid input. For example one where the length field contains a value smaller than `PGSQL_LENGTH_SIZE + PGSQL_PROTO_SIZE`

File: rust/src/pgsql/parser.rs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    }
+}
+
+named!(pub parse_pgsql_startup_packet<PGSQLStartupPacket>,
+    do_parse!(
+        len: bits!(take_bits!(32u32))
+        >> proto_version: bits!(tuple!(
+            take_bits!(16u16),
+            take_bits!(16u16)))
+        >> data: take!(len  as usize - PGSQL_LENGTH_SIZE - PGSQL_PROTO_SIZE)
```
Comment (victorjulien) [inline rust/src/pgsql/parser.rs:50]: I always find this hard to read. Would `len as usize - (PGSQL_LENGTH_SIZE + PGSQL_PROTO_SIZE)` be the same?

**Comment by victorjulien** [comment]
I think this is a great start @jufajardini :+1:

**Comment by victorjulien** [comment]
I think a good next step would be to add tests for incomplete records. Handling these will be an important part of the parser (see https://github.com/OISF/suricata/blob/master/doc/devguide/extending/app-layer/parser.rst for what will be expected when you integrate the parser). While TCP data often gives you a full PDU, it doesn't have to be that way. So we need to handle all cases.

File: rust/src/pgsql/parser.rs (hunk 2 above)
Comment (jufajardini) [reply rust/src/pgsql/parser.rs:50]: I'll try that and see if it works! Although this will likely not be necessary anymore, following Jason's reasoning.

Comment (jufajardini) [reply rust/src/pgsql/parser.rs:50]: I believe it does, I've changed it and code runs, test still working.

**Comment by jufajardini** [comment]
Closed with: https://github.com/OISF/suricata/pull/5706

**Pull request closed without merging**